[tiny released bug] Fleet UI: Fix live query/policy host search icon

RachelElysia opened the pull request:
## Issue
Cerra #16608 

## Description
- Fix search icon bug
- Move CSS into `InputFieldWithIcon`
- Use `InputFieldWithIcon`
- Wrap input field part with a parent wrapper positioned relative and the child icon with in it as position absolute.
- Use display flex, align items center to position any height icon (e.g. caret/email envelope is a smaller height than a magnifying glass)
- Only reposition icon if label exists as icon is being pushed by the label

## Screen recording
https://www.loom.com/share/a6f527349b6f4632a1a9ce3913a9ad5f?sid=86166c00-d0f5-4198-a6af-abfde8132766

# Checklist for submitter

If some of the following don't apply, delete the relevant line.

<!-- Note that API documentation changes are now addressed by the product design team. -->

- [x] Changes file added for user-visible changes in `changes/` or `orbit/changes/`.
- [x] Manual QA for all new/changed functionality

lukeheath commented:
@RachelElysia Maybe accidentally in draft?

jacobshandling commented:
👀

jacobshandling reviewed (dismissed):
This approach causes some issues with SearchFields – maybe just apply this locally, or globally in a different way:

<img width="571" alt="Screenshot 2024-02-23 at 11 19 38 AM" src="https://github.com/fleetdm/fleet/assets/61553566/3aa47a93-25e7-4772-b826-35bd99006b9d">

RachelElysia commented:
Thanks for finding that @jacobshandling -- making a global fix since this has been an ongoing issue.

RachelElysia commented:
Removed position locally for filter icon and added the margin to the wrapper and not just the input

<img width="1609" alt="Screenshot 2024-02-23 at 12 50 45 PM" src="https://github.com/fleetdm/fleet/assets/71795832/61aebfeb-5da2-4865-b459-68d1adf4c603">

RachelElysia commented:
Thanks again for catching, I swept the code and there was no other instances of icons with local styling.
